### ISSUE TITLE
CLI: Incorrect sort gives a stack trace instead of a proper error message

### ISSUE THREAD

**Issue by katiekeel** [kubernetes/kubernetes#104336]
#### What happened:

I asked kubectl to give me a list of nodes sorted by status.

#### What you expected to happen:

I'm not super familiar with the sort syntax, so I expected an error response along the lines of "invalid sort criteria," but instead I got a full stack trace.

#### How to reproduce it (as minimally and precisely as possible):

`kubectl get node -l kubernetes.io/role=node --no-headers --sort-by=status`

#### Anything else we need to know?:

Not sure whether this is really a bug or more of a UI issue but I wanted to report it nonetheless.

#### Environment:
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.11", GitCommit:"c6a2f08fc4378c5381dd948d9ad9d1080e3e6b33", GitTreeState:"clean", BuildDate:"2021-05-12T12:27:07Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.11", GitCommit:"c6a2f08fc4378c5381dd948d9ad9d1080e3e6b33", GitTreeState:"clean", BuildDate:"2021-05-12T12:19:22Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```

- OS (e.g: `cat /etc/os-release`): MacOS Big Sur 11.4

**Comment by wawa0210** [comment]
Can you post the detailed error message? I can't seem to reproduce

```
[root@localhost ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.14", GitCommit:"0fd2b5afdfe3134d6e1531365fdb37dd11f54d1c", GitTreeState:"clean", BuildDate:"2021-08-11T18:07:41Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-21T23:01:33Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
[root@localhost ~]# kubectl get node  --no-headers --sort-by=status
member1-control-plane   Ready   control-plane,master   11d   v1.21.1
```

**Comment by lauchokyip** [comment]
This will be changed for the upcoming `kubectl` 

Same issue:
https://github.com/kubernetes/kubernetes/pull/103458

**Comment by pacoxu** [comment]
@katiekeel As #103458 is merged, you could try the 1.23.0-alpha.0 kubectl.
```
curl -LO "https://dl.k8s.io/release/v1.23.0-alpha.1/bin/linux/amd64/kubectl"

[root@daocloud ~]# ./kubectl get node   --no-headers   --sort-by=status
F0910 15:11:38.500322 2104580 sorter.go:378] Field {.status} in [][][]reflect.Value is an unsortable type: interface, err: unsortable type: map[string]interface {}
[root@daocloud ~]# ./kubectl version
Client Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.0-alpha.1", GitCommit:"3945e35f1cfac9d489242607ddbba8c74c452ec3", GitTreeState:"clean", BuildDate:"2021-08-19T14:43:59Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:39:34Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}

```

/close
as the dup issue is fixed in 1.23.
